pkg/networkdevice/profile/profiledefinition/metrics.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package profiledefinition
+
+import (
+	"regexp"
+)
+
+type ProfileMetricType string
```

Comment:
`configcheck.ProfileMetricType` made sense, but `profiledefinition.ProfileMetricType` stutters a little - couldn't it just be `profiledefinition.MetricType`?

For that matter, would it make sense to rename `profile/profiledefinition` to just `profile/definition`?

Reply:
1/ The intent for this case is to make `ProfileMetricType` more distinct from [Agent MetricType](https://github.com/DataDog/datadog-agent/blob/8914a281cf6f9cfa867e0d72899c39afa51abce7/pkg/metrics/metric_sample.go#L13-L30) since there are have different values.

2/ 

> For that matter, would it make sense to rename profile/profiledefinition to just profile/definition?

The intent is to have it easier to understand when it's taken alone in some circumstances. I mean `profiledefinition` is easier to understand than `definition`. For example `profiledefinition.MetricsConfig` can be easier to understand than `definition.MetricsConfig`.

But agree, it's maybe something for debate :D and adds some verbosity.

Reply:
Thinking about this a little more, could we get rid of `profiledefinitions` entirely and move the code up to `profile`? Then e.g. the metric type would be `profile.MetricType`, which is concise and clear.

FWIW, the recommendation from Effective Go (https://go.dev/doc/effective_go#package-names) is to use one-word package names when possible:

> When a package is imported, the package name becomes an accessor for the contents. After `import "bytes"`
the importing package can talk about `bytes.Buffer`. It's helpful if everyone using the package can use the same name to refer to its contents, which implies that the package name should be good: short, concise, evocative. By convention, packages are given lower case, single-word names; there should be no need for underscores or mixedCaps. Err on the side of brevity, since everyone using your package will be typing that name.

and to minimize repetition:

> The importer of a package will use the name to refer to its contents, so exported names in the package can use that fact to avoid repetition. [...] For instance, the buffered reader type in the bufio package is called Reader, not BufReader, because users see it as bufio.Reader, which is a clear, concise name. Moreover, because imported entities are always addressed with their package name, bufio.Reader does not conflict with io.Reader. Similarly, the function to make new instances of ring.Ring—which is the definition of a constructor in Go—would normally be called NewRing, but since Ring is the only type exported by the package, and since the package is called ring, it's called just New, which clients of the package see as ring.New. Use the package structure to help you choose good names.